components/datetime/src/pattern/runtime/pattern.rs

```diff
@@ -15,26 +15,61 @@ use zerovec::ZeroVec;
     databake(path = icu_datetime::pattern::runtime),
 )]
 #[allow(clippy::exhaustive_structs)] // part of data struct
+#[zerovec::make_varule(PatternULE)]
+#[zerovec::skip_derive(Ord)]
 pub struct Pattern<'data> {
     pub items: ZeroVec<'data, PatternItem>,
     /// This field should contain the smallest time unit from the `items` vec.
     /// If it doesn't, unexpected results for day periods may be encountered.
-    pub time_granularity: TimeGranularity,
+    pub metadata: PatternMetadata,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[zerovec::make_ule(PatternMetadataULE)]
+#[zerovec::skip_derive(Ord)]
+pub struct PatternMetadata(u8);
+
+impl PatternMetadata {
+    #[inline]
+    pub(crate) fn time_granularity(self) -> TimeGranularity {
+        TimeGranularity::from_ordinal(self.0)
+    }
+
+    pub(crate) fn from_items(items: &[PatternItem]) -> Self {
+        let time_granularity: TimeGranularity = items.iter().map(Into::into).max().unwrap_or_default();
+        Self::from_time_granularity(time_granularity)
+    }
+
+    #[inline]
+    pub(crate) const fn from_time_granularity(time_granularity: TimeGranularity) -> Self {
```

Comment:
Observation: databake uses this function but it is a `pub(crate)` function. I want to see if any CI fails.